### DISCUSSION TITLE
Could we have a changelog aligned with Java LTS versions, SDMX versions supported?

### DISCUSSION THREAD

**Discussion by good92** [amattioc/SDMX#242] [General]
Hi,

First thanks.

Could add a changelog that describe JDK/Java LTS requirements, like:

3.0.6  - 11 
2.3.7 - 1.7/1.8  https://github.com/amattioc/SDMX/blob/v2.3.7/JAVA/pom.xml#L50
2.1.2  - 1.7/1.8 https://github.com/amattioc/SDMX/blob/v2.1.2/JAVA/pom.xml#L57

https://endoflife.date/java - Information about Java EOL of 1.8 11 17 - future 20

Information if it support SDMX 2.0, 2.1 and/or 3.0 would be also useful.
Is it just aligned with the current SDMX versioning and SDMX 2.2, 2.3 format exists?

It will help to do the proper migration updates to the different connectors.

I see https://github.com/epam/jsdmx supporting SDMX 3.x, does this repo is link to jsdmx? - They choose Gradle for the build which seems harder to follow than Apache Maven.

Thank you in advance,
regards,

**Comment by amattioc** [comment]
Hi, the current (latest release) requirements for all the various software components are in the wiki: https://github.com/amattioc/SDMX/wiki/System-Requirements 

We build with those software levels (actually only java and R need a compilation, the other connectors are just packaged) and we regularly run tests with the same components.  

Even if the pom mentions java 11, the current java support is still on java 8 (we still use the build.xml for packaging the releases). Of course the code can be probably built with different environments, maybe with small adaptations. We tend to stick to these old versions because many environments of our users are quite conservative and we do not want to prevent anyone from the possibility of using the tool.

Regarding the SDMX versions, this is a bit more complex. We currently support, seamlessly, providers that implement the SDMX 2.1 messages and API. For some providers that are on SDMX v2.0 (or light variations of it) we developed custom providers. We started introduucing support for the REST API v.2 (related to SDMX 3.0) but we still do not manage 3.0 messages. In any case I don't think there are real providers that implement sdmx 3.0 yet.

If you don't mind I'm converting this issue to a discussion. I'll be glad to have suggestions on how to better manage this project.
